server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java

```diff
@@ -36,38 +41,52 @@
 
     private volatile boolean destructiveRequiresName;
 
+    private volatile boolean isDestructiveRequiresNameSet;
+
     public DestructiveOperations(Settings settings, ClusterSettings clusterSettings) {
         destructiveRequiresName = REQUIRES_NAME_SETTING.get(settings);
+        isDestructiveRequiresNameSet = REQUIRES_NAME_SETTING.exists(settings);
         clusterSettings.addSettingsUpdateConsumer(REQUIRES_NAME_SETTING, this::setDestructiveRequiresName);
     }
 
     private void setDestructiveRequiresName(boolean destructiveRequiresName) {
+        this.isDestructiveRequiresNameSet = true;
         this.destructiveRequiresName = destructiveRequiresName;
     }
 
     /**
      * Fail if there is wildcard usage in indices and the named is required for destructive operations.
      */
     public void failDestructive(String[] aliasesOrIndices) {
-        if (destructiveRequiresName == false) {
-            return;
-        }
-
         if (aliasesOrIndices == null || aliasesOrIndices.length == 0) {
-            throw new IllegalArgumentException("Wildcard expressions or all indices are not allowed");
+            handleWildcardOperation();
         } else if (aliasesOrIndices.length == 1) {
             if (hasWildcardUsage(aliasesOrIndices[0])) {
-                throw new IllegalArgumentException("Wildcard expressions or all indices are not allowed");
+                handleWildcardOperation();
             }
         } else if (Arrays.equals(aliasesOrIndices, MATCH_NONE_PATTERN) == false) {
             for (String aliasesOrIndex : aliasesOrIndices) {
                 if (hasWildcardUsage(aliasesOrIndex)) {
-                    throw new IllegalArgumentException("Wildcard expressions or all indices are not allowed");
+                    handleWildcardOperation();
                 }
             }
         }
     }
 
+    private void handleWildcardOperation() {
```

Comment:
nit: Perhaps a name like "checkWildcardOk" or something along those lines would make more sense? Handle makes me think this is doing some work regarding wildcards, but it is just checking deprecations/errors.

Reply:
That is a better name than what I used. I'll change it.